### PR TITLE
Make config file loading compatible with JSCS spec

### LINKED ISSUE #85: Feature request: Configure what configuration file to look for
I have an issue with the way `linter-jscs` looks for configuration file. 

In our project, we have something similar to the following structure:

     root
        / subproject
            - package.json
            / node_modules
            / app
        / second subproject
            - package.json
            / node_modules
            / app

We want to enforce the same styles for all our JavaScript across the entire project by having a `.jscsrc` file on the project root.

     root
        - .jscsrc     <-- global style rules
        / subproject
            - package.json
            / node_modules
            / app
        / second subproject
            - package.json
            / node_modules
            / app

When checking styles from command line using `jscs ./subproject/app`, jscs will use the right configuration. But inside atom, it will always pick up on the package.json file in the subproject when checking files inside the app folders.

So I wan't a way to configure myself out of this issue.

Today, linter-jscs have a fixed set of config files that it search for:

        const configFiles = ['.jscsrc', '.jscs.json', 'package.json'];

        // Search for project config file
        let config = findFile(filePath, configFiles);

If the configFiles-array could be a configuration option, I could remove package.json from the list.

### PR DESCRIPTION
**Pull request by acusti**
Checks for `.jshintrc`, then `.jscs.json`, then the `jscsConfig` property in `package.json`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
 'use babel';
 
 import JSCS from 'jscs';
-import { Range } from 'atom';
-import { findFile } from 'atom-linter';
-import { readFileSync } from 'fs';
-import stripJSONComments from 'strip-json-comments';
+import path from 'path';
+import configFile from './node_modules/jscs/lib/cli-config';
```
Comment (acusti) [inline index.js]: @iam4x Importing the module from the JSCS package like this seems terrible. Do you know a better way?

Comment (AnimiVulpis) [inline index.js]: @acusti I learned that it should work like this:
`import configFile from 'jscs/lib/cli-config'`
_as learned here:_
https://github.com/jscs-dev/node-jscs/issues/1673

Comment (acusti) [inline index.js]: @AnimiVulpis Nice! Thanks for the research. I didn’t even think to try that.

**Comment by acusti** [comment]
@iam4x This seems pretty ready for prime time, though it did wind up taking on more scope than I originally intended. If you’d prefer, I can create separate PRs for each change in functionality. On the other hand, I made atomic commits, so the [list of commits](https://github.com/AtomLinter/linter-jscs/pull/87/commits) does a pretty good job of breaking down each separate issue being addressed.

**Comment by acusti** [comment]
@iam4x I’ve been using this branch for a while now and it works well. There is one issue: a4cdeed421cd013c6ffdf20647375d765ea47894, the commit that forces `fixString` to run after the linter, also means that `fixString` runs after the editor save event occurs, so the changes made by fixString don’t get saved. The dumb fix would be to just save the editor again, but that sucks. The behavior can also be left as is; the benefit is that it makes it easy for the user to evaluate the results of `fixString` before saving it.

I don’t know how to make the order in which the listeners are triggered deterministic, and I’m not familiar with how the linter attaches its event listeners, so that’s all I’ve got for now. Would be nice to get this merged, though.

**Comment by iam4x** [comment]
@acusti Thank's for the PR!

I was on vacation, came back today and I have tons of stuff at work... I hope I can get this merged before the weekend :+1:

**Comment by acusti** [comment]
Great! Hope you had a great vacation, that I wasn’t too pushy. No more pestering at least until next week :wink:

**Comment by blah238** [comment]
:+1: LGTM

**Comment by abhishekdev** [comment]
Any updates on when this might get merged? Eagerly waiting for this one.

/cc @iam4x

**Comment by iam4x** [comment]
@acusti Could you rebase on master? I will merge it :+1:

**Comment by acusti** [comment]
@iam4x I merged the upstream changes. I think I caught everything.

**Comment by iam4x** [comment]
Sorry for being so long :(

**Comment by justinaray** [comment]
Woot!  Thanks for the merge and all the work on this one.  Been waiting for this so I can use JSCS in my IDE and build pipeline.  Build pipeline uses non-standards, so I can't share files and needed an override!

**Pull request merged**